src/qibolab/result.py

```diff
@@ -97,26 +71,26 @@ def to_dict_probability(self, state=1):
         elif state == 0:
             return {"probability": self.ground_state_probability}
 
-    def compute_average(self):
+    @property
+    def average(self):
         """Perform average over i and q"""
-        return AveragedResults(self.array.i.mean(), self.array.q.mean())
+        return AveragedResults.from_components(np.mean(self.i), np.mean(self.q))
 
-    def to_dict(self, average=True):
-        """Serialize output in dict.
-        Args:
-            average (bool): If `True` returns a dictionary of the form
-                            {'MSR[V]' : v, 'i[V]' : i, 'q[V]' : q, 'phase[rad]' : phase}.
-                            Where each value is averaged over the number shots. If `False`
-                            all the values for each shot are saved.
-        """
-        results = self.compute_average() if average else self
+    @property
+    def serial(self):
+        """Serialize output in dict."""
 
         return {
-            "MSR[V]": np.sqrt(results.i**2 + results.q**2),
-            "i[V]": results.i,
-            "q[V]": results.q,
-            "phase[rad]": np.angle(results.i + 1.0j * results.q),
+            "MSR[V]": self.measurement,
+            "i[V]": self.i,
+            "q[V]": self.q,
+            "phase[rad]": self.phase,
         }
 
     def __len__(self):
+        assert len(self.i) == len(self.q)
```

Comment:
I believe it would make more sense to have this check in the constructors and raise a proper error if this condition is not satisfied, rather than here. Also given that `i` and `q` are always arrays, we could also check their whole shape.